src/services/item/plugins/action/utils.ts

```diff
@@ -8,6 +12,44 @@ export enum ItemActionType {
   Move = 'move',
 }
 
+export const validateAggregateRequest = (
+  countGroupBy: AggregateAttribute[],
+  aggregateFuction: AggregateFunctionType,
+  aggregateMetric: AggregateAttribute,
+  aggregateBy: AggregateAttribute[],
+) => {
+  // Aggregate by user is not allowed
+  if (aggregateBy?.includes('user')) {
+    return false;
+  }
+
+  // Perform aggregation on a grouping expression is not allowed
+  if (aggregateBy?.includes(aggregateMetric)) {
+    return false;
+  }
+
+  // The input of the second stage aggregation should be the output of the first stage aggregation
+  if (!(countGroupBy.includes(aggregateMetric) || aggregateMetric === 'actionCount')) {
+    return false;
+  }
+  let falseFlag = false;
+  aggregateBy?.forEach((element) => {
+    if (!(countGroupBy.includes(element) || element === 'actionCount')) {
+      falseFlag = true;
```

Comment:
Throw an error, then you won't even need to use `falseFlag`.
I personally prefer when validate throw rather than returning a boolean. If you throw, the endpoint should automatically throw as well (don't need to manually use reply). You can get inspiration from `action/utils/errors.ts` file.